### ISSUE TITLE
'async for' requires an object with __aiter__ method, got s3.Bucket.objectsCollection

### ISSUE THREAD

**Issue by parinmaru2** [terrycain/aioboto3#188]
* Async AWS SDK for Python version: aioboto3-6.3.0
* Python version: Python3.6
* Operating System: Linux

**Problem:**

I am using aioboto3 to create s3 resource.
And then I call s3 objects filter method:

```
my_bucket = self.s3_resource.Bucket(bucket_name)
response = my_bucket.objects.filter(Prefix=remote_directory_prefix)
```
I can't loop/iterate over this response:

```
async for r in response:
```
Above async for throws this exception:
**'async for' requires an object with aiter method, got s3.Bucket.objectsCollection**

**Analysis**
It seems s3.Bucket.objectsCollection doesn't implement aiter() method

**Comment by terrycain** [comment]
Think this'll be solved by either #181 or #186

**Comment by terrycain** [comment]
Should be fixed by v6.5.0 can you test it out? If it doesn't work can you reopen this ticket with the issue.

**Comment by parinmaru2** [comment]
> Should be fixed by v6.5.0 can you test it out? If it doesn't work can you reopen this ticket with the issue.

Thanks @terrycain 
Tested with latest version and this seems to have been fixed. Thanks again.
